File: curtain-server/src/main/java/com/coisini/curtain/exception/basic/HttpException.java
package com.coisini.curtain.exception.basic;

/**
 * @Description Http异常类
 * @author coisini
 * @date Aug 10, 2021
 * @Version 1.0
 */
public class HttpException extends RuntimeException{

    protected Integer code;
    protected Integer httpStatusCode = 500;

    public Integer getCode() {
        return code;
    }

    public Integer getHttpStatusCode() {
        return httpStatusCode;
    }
}

File: curtain-store/core/http-exception.js
/**
 * @Description HTTP Exception
 * @author coisini
 * @date Sep 2, 2021
 * @Version 1.0
 */
export class HttpException extends Error {

    errorCode = 9999
    statusCode = 500
    message = ''

    constructor(errorCode, message, statusCode) {
        super()
        this.message = message;
        this.errorCode = errorCode;
        this.statusCode = statusCode;
    }

}
